.github/workflows/push.yml

```diff
@@ -0,0 +1,49 @@
+name: Single Build & Test (Push)
+
+# workflow is for a branch push only. do not run on push to master.
+# push to master (which is a pull requet merge) has a more elaborate workflow to run
+# github repo is configured with branch protection on master.
+on: 
+  push:
+    branches-ignore:
+      - 'master'
+
+  workflow_dispatch:
+    inputs: 
+      node:
+        required: false
+        description: 'A build enabled node image (e.g: node:15.14.0) or docker dir (e.g. ./.github/docker-node/16/centos7-build)'
+        default: 'node' # as of June 2021 this is node:16-buster
+
+jobs:
+  build-single-test:
+    runs-on: ubuntu-latest # environment job will run in
+
+    env:
+      AO_TOKEN_PROD: ${{ secrets.AO_TOKEN_PROD }}
+
+    steps:
+      - name: Checkout ${{ github.ref }}
+        uses: actions/checkout@v2
+
+      - name: Create Container ${{ github.event.inputs.node || 'node' }}
+        id: create
+        run: .github/scripts/container-from-matrix-input.sh ${{ github.event.inputs.node || 'node' }} # when running from 'push' there is no input, default to 'node' image
+
+      # must install specific dependency for liboboe.js before a build
+      - name: NPM Install linux-os-info
+        run: docker exec ${{ steps.create.outputs.containerId }} npm install --save linux-os-info --unsafe-perm
```

Comment:
Lin,thanks for catching this! It's a gotcha and it was left 'half-baked' when it should have been revisited. My bad.

The logic:

1. We must install before rebuild. For `linux-os-info` but also for `node-pre-gyp` and all other dependencies.

2. `npm install --unsafe-perm` will run `node setup-liboboe.js && node-pre-gyp install --fallback-to-build`. This is a `node-pre-gyp` configuration requirement (see: https://github.com/mapbox/node-pre-gyp#configuring) with our pre-addition. Let's call this *user install script*.

3. If we run the *user install script*, and the package has never been built before (e.g. v11.1.2-34) `node-pre-gyp` won't find a pre-build tarball and will fallback-to-build from source *at the install stage* (placing result in `./dist`). Running `npm run rebuild` in the following command will rebuild package *again* (placing result again in `./dist`).

4. To avoid that (without YAML conditionals and bucket calls) we want to avoid the *user install script* with our `npm` command.

5. `npm install linux-os-info @mapbox/node-pre-gyp node-addon-api nan --unsafe-perm --no-package-lock` will work but the flag is deprecated in NPM 7 (see: https://github.com/npm/cli/issues/3023). No good. Also in order to run the tests we'll need a full install two commands down anyway (other note: `node-gyp` is in dependencies, but can build without).

5. `npm install --unsafe-perm --ignore-scripts` will, in our current stats, get us what we want, but the meaning of the flag is to ignore *all scripts* not just *our scripts* (see here: https://npm.community/t/ignore-scripts-ignores-all-scripts/1403). Not best.

7. `npm install linux-os-info --unsafe-perm` tells npm to install that single package *and* everything else listed in package.json but *bypasses* the *user install script*. That's the result we want.


The fix:

1. Remove --save form `npm install` (obviously) not needed.

2. Remove --unsafe-perm from `npm run rebuild` not needed.

3. Naming and comment:
```
      # must install specific dependencies before a build
      # can't call npm install. doing so may fallback-to-build if packed has yet to be published (double build)
      # use npm workaround specifying a package name to bypass install script in package.json
      - name: NPM Install dependencies
        run: docker exec ${{ steps.create.outputs.containerId }} npm install linux-os-info --unsafe-perm
```

Thoughts?

Reply:
Good find on the gotcha with `--ignore-scripts`, option 7 and the updated step naming/comment lgtm.

Reply:
Done.